src/main/java/dev/waterdog/waterdogpe/network/downstream/ConnectedDownstreamHandler.java

```diff
@@ -45,6 +47,28 @@ public final boolean handle(RemoveObjectivePacket packet) {
         return false;
     }
 
+    @Override
+    public final boolean handle(SetScorePacket packet) {
+        switch(packet.getAction()) {
+            case SET:
+                for(ScoreInfo info : packet.getInfos()) {
+                    this.player.getScoreInfos().add(info);
+                }
+                break;
+            case REMOVE:
+                for(ScoreInfo info : packet.getInfos()) {
```

Comment:
Since `scoreInfos` is set, we can just do `scoreInfos.remove()`

Reply:
remove would require the ScoreInfo to be the same they wont be the same (or equal) you wont be sending the same ScoreInfo for setting (needs more stuff like entity id) and removing (does not contain entity id -> entityId = -1) xd

Reply:
Than store them in map by scoreboard id. So we dont need to create extra iterator each time.

Reply:
Changed